apps/src/templates/teacherHomepage/AnnouncementsCarousel.jsx

```diff
@@ -0,0 +1,78 @@
+import React from 'react';
+import FontAwesome from '../FontAwesome';
+import color from "../../util/color";
+import _ from 'lodash';
+
+const styles = {
+  box: {
+    width: 940,
+    backgroundColor: color.white
+  },
+  arrowIcon: {
+    fontSize: 40,
+    lineHeight: "40px",
+    color: color.white,
+    marginRight: 15,
+    marginLeft: 15
+  },
+  arrowBox: {
+    float: "right",
+    marginTop: 12,
+    marginRight: 25,
+    height: 40,
+    width: 90,
+    borderColor: color.white,
+    borderWidth: 1,
+    borderRadius: 3,
+    borderStyle: "solid"
+  },
+  arrowBoxRight: {
+    marginTop: 25,
+    marginRight: 25
+  },
+};
+
+const AnnouncementsCarousel = React.createClass({
+  propTypes: {
+    children: React.PropTypes.oneOfType([
+      React.PropTypes.node,
+      React.PropTypes.arrayOf(React.PropTypes.node)
+    ])
+  },
+
+  getInitialState() {
+    return {currentIndex: 0};
+  },
+
```

Comment:
Rather than having `() => this.seeAnother(-1)`, I would just create two new class methods (something like clickPrevious, clickNext). The reason for this is somewhat subtle (and also a little bit nitty).

The way you have things right now, we'll create two new methods every time we render the announcement. If we instead have them as sibling methods to seeAnother, they will only get created once per instance.

For this particular component, we're probably not going to render it enough times for us to be truly concerned about this, but it's a good pattern to avoid. One could conceive of React components that get updated very frequently where having the method creation happen in the render method leads to undue memory pressure.